nestor_api/lib/io.py

```diff
@@ -0,0 +1,102 @@
+"""I/O library"""
+
+import errno
+import math
+import os
+import shutil
+import subprocess
+
+from datetime import datetime
+from pathlib import Path
+from random import random
+
+import yaml as yaml_lib
+
+from nestor_api.config.config import Configuration
+
+
+def copy(source, destination):
+    """Copy file or directory with its content from source to destination"""
+    try:
+        shutil.copytree(source, destination)
+    except OSError as err:
+        if err.errno == errno.ENOTDIR:
+            shutil.copy(source, destination)
+        else:
+            raise
+
+
+def create_temporary_directory(directory_prefix=""):
+    """Creates a temporary directory with an optional prefix"""
+    tmp_directory_path = get_temporary_directory_path(directory_prefix)
+
+    # Create directory if it does not exist
+    ensure_dir(tmp_directory_path)
+
+    return tmp_directory_path
+
+
+def ensure_dir(directory_path):
+    """Ensures that a directory exists, else creates it"""
+    Path(directory_path).mkdir(mode=0o777, parents=True, exist_ok=True)
+
+
+def execute(command: str, cwd=None):
+    """Executes a command and returns the stdout from it"""
+    result = subprocess.run(command.split(), stdout=subprocess.PIPE, check=True, cwd=cwd)
```

Comment:
What about stderr ? I would pipe it also, isn't it ?

Reply:
The thing is that if this returns a not zero exit code, this instruction will throw (check=True). So I am not sure of the use case where we would need to pipe stderr.